OpenRA.Mods.Common/Activities/Air/ReturnToBase.cs

```diff
@@ -111,12 +112,24 @@ public override Activity Tick(Actor self)
 					return new FlyCircle(self);
 			}
 
-			return ActivityUtils.SequenceActivities(
-				new Fly(self, Target.FromPos(w1)),
-				new Fly(self, Target.FromPos(w2)),
-				new Fly(self, Target.FromPos(w3)),
-				new Land(self, Target.FromActor(dest)),
-				NextActivity);
+			List<Activity> landingProcedures = new List<Activity>();
+
+			var turnRadius = CalculateTurnRadius(planeInfo.Speed);
+
+			landingProcedures.Add(new Fly(self, Target.FromPos(w1), new WDist(0), new WDist(turnRadius * 3)));
+			landingProcedures.Add(new Fly(self, Target.FromPos(w2)));
+
+			// Fix a problem when the airplane is send to resupply near the airport
+			landingProcedures.Add(new Fly(self, Target.FromPos(w3), new WDist(0), new WDist((int)(turnRadius / 2))));
```

Comment:
Nit: `WDist.Zero` instead of `new WDist(0)`.

Also, since `turnRadius` already is an `int`, this `(int)` should be unnecessary.